### PR TITLE
Update Status API route with tests

### PR DESCRIPTION
**Pull request by btotharye**
**Proposed changes**:
- Added in test for previous merge that updated the /status api route to give back the true model directory and file and not the tmp directory. This test ensures we get back the proper model dir and not the tmp location.

**Status (please check what you already did)**:
- [X] made PR ready for code review
- [X] added some tests for the functionality
- [ ] updated the documentation
- [X] updated the changelog
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by btotharye** [comment]
still looking at this @wochinge since my local tests are now failing due to the model location:

```
    def test_status(rasa_app: SanicTestClient):
        _, response = rasa_app.get("/status")
        model_file = response.json["model_file"]
        assert response.status == 200
        assert "fingerprint" in response.json
        assert "model_file" in response.json
>       assert "models/" in model_file
E       AssertionError: assert 'models/' in '/private/var/folders/x5/pkbj85f136zg0kr17ghzlhl40000gn/T/pytest-of-bhopkins/pytest-69/models0/20191111-095000.tar.gz'

tests/test_server.py:111: AssertionError
```

I'm using the same code I had on the master version but it seems to be doing different stuff here

**Comment by wochinge** [comment]
> assert 'models/'

you could also just ensure that the model file is the same as the one from the `trained_rasa_model` fixture?

File: CHANGELOG.rst [hunk 1]
```diff
@@ -21,6 +21,7 @@ Changed
 - Updated Mattermost connector documentation to be more clear.
 - Updated format strings to f-strings where appropriate.
 - Updated tensorflow requirement to ``1.15.0``
+- Updated`/status` endpoint to show correct path to model archive
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
- Updated  ``/status`` endpoint to show correct path to model archive
```

**Comment by tmbo** [comment]
@btotharye can you please take another look at this PR?

**Comment by btotharye** [comment]
> @btotharye can you please take another look at this PR?

Yea sorry got busy the past couple days with customer issues working on finalizing it today

**Comment by tmbo** [comment]
Yes no worries, just want to make sure it doesn't go stale :+1:

**Comment by btotharye** [comment]
@wochinge this new regex thing I did seems to work, not sure if its hacky or a better way but the issue is the tests have a different folder layout it does from the normal server running if that makes sense.

File: CHANGELOG.rst [hunk 2]
```diff
@@ -45,6 +45,10 @@ Changed
 - Updated Mattermost connector documentation to be more clear.
 - Updated format strings to f-strings where appropriate.
 - Updated tensorflow requirement to ``1.15.0``
+- Updated ``/status`` endpoint to show correct path to model archive
```
Comment (wochinge) [inline CHANGELOG.rst]: please move this to the correct position

File: rasa/core/agent.py [hunk 3]
```diff
@@ -902,6 +906,7 @@ def load_local_model(
 
         working_directory = tempfile.mkdtemp()
         unpacked_model = unpack_model(model_archive, working_directory)
+        print(f"The model_archive is: {model_archive}")
```
Comment (wochinge) [inline rasa/core/agent.py]:
```suggestion
```

File: tests/test_server.py [hunk 4]
```diff
@@ -104,13 +105,17 @@ def test_version(rasa_app: SanicTestClient):
 
 def test_status(rasa_app: SanicTestClient):
     _, response = rasa_app.get("/status")
+    model_file = response.json["model_file"]
+    test_folder_pattern = re.search(r"models\d/\d+-\d+.tar.gz", model_file)
     assert response.status == 200
     assert "fingerprint" in response.json
     assert "model_file" in response.json
+    assert "models/" in model_file or test_folder_pattern
```
Comment (wochinge) [inline tests/test_server.py]: can you not test `os.path.isfile`? What problem did you run into so that you had to use the regex ?

File: CHANGELOG.rst (hunk 2 above)
Comment (btotharye) [reply CHANGELOG.rst]: sorry where should this go again?  I had this originally there from when we pushed the first changes so should I just move this up or say added tests in under the most recent changelog section?

File: tests/test_server.py (hunk 4 above)
Comment (btotharye) [reply tests/test_server.py]: I just forgot about it honestly so I updated it back to check for that.

File: tests/test_server.py [hunk 5]
```diff
@@ -104,13 +105,16 @@ def test_version(rasa_app: SanicTestClient):
 
 def test_status(rasa_app: SanicTestClient):
     _, response = rasa_app.get("/status")
+    model_file = response.json["model_file"]
     assert response.status == 200
     assert "fingerprint" in response.json
     assert "model_file" in response.json
+    assert "models/" in model_file or os.path.isfile(response.json["model_file"])
```
Comment (wochinge) [inline tests/test_server.py]: so this doesn't work?
```suggestion
    os.path.isfile(response.json["model_file"])
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -45,6 +50,9 @@ Changed
 - Updated Mattermost connector documentation to be more clear.
 - Updated format strings to f-strings where appropriate.
 - Updated tensorflow requirement to ``1.15.0``
+
```
Comment (wochinge) [inline CHANGELOG.rst]: that should probably not go in

File: CHANGELOG.rst [hunk 7]
```diff
@@ -31,6 +31,11 @@ Fixed
 - Fixed bug that stopped Dockerfiles from building version 1.4.4.
 - Fixed format guessing for e2e stories with intent restated as ``/intent``
 
+Changed
```
Comment (wochinge) [inline CHANGELOG.rst]: there were some releases in between. you need to update the location in the changelog

File: tests/test_server.py (hunk 5 above)
Comment (btotharye) [reply tests/test_server.py]: I'll check it again I don't think it did I need to remember why, I'll check it when I get home later from traveling.  I also thought we wanted to also verify the proper model dir part as well in the response or no?

Comment (btotharye) [reply tests/test_server.py]: It does work with just the isfile check but shouldn't we also verify the path we get back is also the `models/model.tar.gz` and not the tmp location?

File: CHANGELOG.rst (hunk 7 above)
Comment (btotharye) [reply CHANGELOG.rst]: can you see if my recent update is right?  If not we can talk about it tomorrow on our call to ensure I have the changelog right.

**Comment by btotharye** [comment]
ok @wochinge just cleaned up the last linting issue all tests should pass shortly

File: CHANGELOG.rst [hunk 8]
```diff
@@ -15,6 +15,8 @@ Added
 
 Changed
 -------
+- Updated ``/status`` endpoint to show correct path to model archive
+
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
```

File: CHANGELOG.rst [hunk 9]
```diff
@@ -33,6 +35,7 @@ Fixed
 - Fixed bug that stopped Dockerfiles from building version 1.4.4.
 - Fixed format guessing for e2e stories with intent restated as ``/intent``
 
+
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
```

File: CHANGELOG.rst [hunk 10]
```diff
@@ -47,6 +50,9 @@ Changed
 - Updated Mattermost connector documentation to be more clear.
 - Updated format strings to f-strings where appropriate.
 - Updated tensorflow requirement to ``1.15.0``
+
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
```

File: CHANGELOG.rst [hunk 11]
```diff
@@ -47,6 +50,9 @@ Changed
...
+Removed
```
Comment (wochinge) [inline CHANGELOG.rst]: please don't change the changelog 🙂 
```suggestion
```

File: CHANGELOG.rst [hunk 12]
```diff
@@ -47,6 +50,9 @@ Changed
...
+-------
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
```

File: rasa/server.py [hunk 13]
```diff
@@ -442,7 +443,7 @@ def create_app(
         except Exception as e:
             logger.debug(traceback.format_exc())
             raise ErrorResponse(
-                500, "ConversationError", f"An unexpected error occurred. Error: {e}",
+                500, "ConversationError", f"An unexpected error occurred. Error: {e}"
```
Comment (wochinge) [inline rasa/server.py]: can you please make sure you have the same black version as we install in travis (black==19.10b0)? This caused a lot of unnecessary diffs

File: tests/test_server.py [hunk 14]
```diff
@@ -102,15 +103,18 @@ def test_version(rasa_app: SanicTestClient):
...
     assert response.status == 200
     assert "fingerprint" in response.json
-    assert "model_file" in response.json
+    assert os.path.isfile(model_file)
+    assert model_file == trained_rasa_model
 
 
 def test_status_nlu_only(rasa_app_nlu: SanicTestClient):
     _, response = rasa_app_nlu.get("/status")
+    assert os.path.isfile(response.json["model_file"])
```
Comment (wochinge) [inline tests/test_server.py]: you can have the same assert as above (`assert model_file == trained_rasa_model`)

File: tests/test_server.py [hunk 15]
```diff
@@ -2,6 +2,7 @@
 import time
 import tempfile
 import uuid
+import re
```
Comment (wochinge) [inline tests/test_server.py]:
```suggestion
```

File: rasa/server.py (hunk 13 above)
Comment (btotharye) [reply rasa/server.py]: yea I don't know why that happened I installed it from the requirements file which I thought is what travis was using as well, I've updated my venv tho.

**Comment by wochinge** [comment]
@btotharye Tests failed, can you please check?

**Comment by btotharye** [comment]
> @btotharye Tests failed, can you please check?

Yea apparently I can't do this??

```
def test_status_nlu_only(rasa_app_nlu: SanicTestClient, trained_rasa_model: Text):
        _, response = rasa_app_nlu.get("/status")
        model_file = response.json["model_file"]
>       assert model_file == trained_rasa_model
E       AssertionError: assert '/private/var/folders/x5/pkbj85f136zg0kr17ghzlhl40000gn/T/pytest-of-bhopkins/pytest-3/models3/nlu-20191122-100419.tar.gz' == '/private/var/folders/x5/pkbj85f136zg0kr17ghzlhl40000gn/T/pytest-of-bhopkins/pytest-3/models4/20191122-100419.tar.gz'
```

Looking into it after this call.

**Comment by btotharye** [comment]
So looks like something wonky is going on now @wochinge with the `trained_rasa_model` info versus what status gives back:


```
E       AssertionError: assert '/private/var...105903.tar.gz' == '/private/var...105904.tar.gz'
E         - /private/var/folders/x5/pkbj85f136zg0kr17ghzlhl40000gn/T/pytest-of-bhopkins/pytest-5/models0/nlu-20191122-105903.tar.gz
E         ?                                                                                            ^ ----              ^
E         + /private/var/folders/x5/pkbj85f136zg0kr17ghzlhl40000gn/T/pytest-of-bhopkins/pytest-5/models1/20191122-105904.tar.gz
E         ?
```

**Comment by btotharye** [comment]
@wochinge I reverted back to the way I had it for now, thoughts on just going this way until we get this NLU retrain issue figured out?  You should see the comment I made in the file with the updated way to do it once we fix the issue.

File: tests/test_server.py [hunk 16]
```diff
@@ -102,18 +102,25 @@ def test_version(rasa_app: SanicTestClient):
...
+    model_file = response.json["model_file"]
     assert response.status == 200
     assert "fingerprint" in response.json
-    assert "model_file" in response.json
+    assert os.path.isfile(model_file)
+    assert model_file == trained_rasa_model
 
 
-def test_status_nlu_only(rasa_app_nlu: SanicTestClient):
+def test_status_nlu_only(rasa_app_nlu: SanicTestClient, trained_rasa_model: Text):
```
Comment (wochinge) [inline tests/test_server.py]: there should be some `trained rasa nlu model` fixture.

**Comment by wochinge** [comment]
@btotharye we won't make it for the 1.5 release, so as soon as the new release is cut, could you rebase your change to go against `1.5.x` and we will include in 1.5.1 ?

**Comment by btotharye** [comment]
> @btotharye we won't make it for the 1.5 release, so as soon as the new release is cut, could you rebase your change to go against `1.5.x` and we will include in 1.5.1 ?

Yea sorry for the delay

File: tests/test_server.py (hunk 16 above)
Comment (btotharye) [reply tests/test_server.py]: I'm still looking for it I don't see anything around the same name like `trained_rasa_nlu` for example searching anywhere.

Comment (btotharye) [reply tests/test_server.py]: found it, was `trained_nlu_model` and its passing now, will rebase

**Comment by btotharye** [comment]
> @btotharye we won't make it for the 1.5 release, so as soon as the new release is cut, could you rebase your change to go against `1.5.x` and we will include in 1.5.1 ?

I don't see a `1.5.x` yet should I be creating this or what is the protocol there?  I have my tests passing now.

File: CHANGELOG.rst [hunk 17]
```diff
@@ -47,6 +48,9 @@ Changed
 - Updated Mattermost connector documentation to be more clear.
 - Updated format strings to f-strings where appropriate.
 - Updated tensorflow requirement to ``1.15.0``
+
```
Comment (wochinge) [inline CHANGELOG.rst]: what about this line?
```suggestion
```

File: CHANGELOG.rst [hunk 18]
```diff
@@ -47,6 +48,9 @@ Changed
...
+Removed
```
Comment (wochinge) [inline CHANGELOG.rst]: why did you add this section?
```suggestion
```

File: CHANGELOG.rst [hunk 19]
```diff
@@ -47,6 +48,9 @@ Changed
...
+-------
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
```

File: tests/test_server.py (hunk 16 above)
Comment (wochinge) [inline tests/test_server.py]: you need this fixture
```suggestion
def test_status_nlu_only(rasa_app_nlu: SanicTestClient, trained_nlu_model: Text):
```

File: CHANGELOG.rst (hunk 17 above)
Comment (btotharye) [reply CHANGELOG.rst]: not sure if you had the latest so I just pushed up everything I had so this is probably old now.

**Comment by btotharye** [comment]
@wochinge can you double check I don't think these Travis errors are related to my code just wanted to know what I have left to do if anything before merging this one.

**Comment by wochinge** [comment]
@btotharye Re-opened your pr

File: CHANGELOG.rst [hunk 20]
```diff
@@ -77,6 +78,9 @@ Changed
 - Updated Mattermost connector documentation to be more clear.
 - Updated format strings to f-strings where appropriate.
 - Updated tensorflow requirement to ``1.15.0``
+
+Removed
```
Comment (wochinge) [inline CHANGELOG.rst]: @btotharye Please don't resolve these comments unless you fix them! Can you please remove these added lines so that we can merge this then?

Comment (btotharye) [reply CHANGELOG.rst]: I thought I had fixed them locally, I'll double check it

Comment (btotharye) [reply CHANGELOG.rst]: ok this should now be resolved sorry, there was some confusion on my end with the changelog because I didn't add this changes I pulled in did somewhere along the line.

Thanks

**Comment by btotharye** [comment]
@wochinge the test coverage is lower due to ` /rasa/nlu/training_data/training_data.py` so does that mean I can merge this or what is the protocol here?

**Comment by wochinge** [comment]
Lets merge!!!!

**Comment by btotharye** [comment]
yay lol
